webClient/src/app/shared/http/http.data.adapter.service.ts

```diff
@@ -97,9 +97,10 @@ export class DataAdapterService {
     };
   }
 
-  convertDatasetContent(responseData: any): { contents: string } {
+  convertDatasetContent(responseData: any): { contents: string, etag?: string } {
     return {
-      contents: JSON.parse(responseData).records.join("\n")
+      contents: JSON.parse(responseData).records.filter(function(record){return record.length > 0}).map(function(record){return record.trim()}).join("\n"),
+      etag: JSON.parse(responseData).etag
```

Comment:
You set etag as an optional arg but you're hardcoding etag as an attribute of the returned object. Not sure if this is intentional, if it is, then the args declaration should change (for example, if you have checks in places for where etag is an attribute but is undefined or null etc.) OR if it truly is optional, then you should conditionally add it as an attribute to the object

something like this should work,
```
const myObj = {
    name: this.username,
    DOB: new Date(this.inputDate),
    ...(this.userGender && { gender : this.userGender })
}
```